.github/workflows/release.yml

```diff
@@ -0,0 +1,47 @@
+name: Release
+
+on:
+  pull_request:
+    types: [closed]
+    branches:
+      - master
+
+jobs:
+  Release:
+    name: 🚀 Release
+    if: github.event.pull_request.merged && startsWith(github.head_ref, 'release-')
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v2
+        with:
+          fetch-depth: 0
+  
+      - uses: actions/github-script@v5
+        with:
+          script: |
+            const get_change_log_diff = require('./scripts/get_changelog_diff.js')
+            core.exportVariable('CHANGELOG', get_change_log_diff())
+
+            // Getting the release version from the PR source branch
+            // Source branch looks like this: release-1.0.0
+            const version = context.payload.pull_request.head.ref.split('-')[1]
+            core.exportVariable('VERSION', version)
+
+      - name: Set up Go
+        uses: actions/setup-go@v2
+        with:
+          go-version: "1.18"
+
+      - name: Build artifacts
+        run: |
+          GOOS=darwin go build -o $PWD/darwin ./cmd/stream-cli
+          GOOS=linux go build -o $PWD/linux ./cmd/stream-cli
+          GOOS=windows go build -o $PWD/windows.exe ./cmd/stream-cli
+
+      - name: Create release on GitHub
+        uses: ncipollo/release-action@v1
+        with:
+          body: ${{ env.CHANGELOG }}
+          tag: ${{ env.VERSION }}
+          token: ${{ secrets.GITHUB_TOKEN }}
+          artifacts: "darwin,linux,windows.exe"
```

Comment:
I want to change this to [goreleaser](https://goreleaser.com/) a bit later, but let's just keep it this way for now

Reply:
sounds good